src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -284,7 +286,9 @@ namespace winrt::TerminalApp::implementation
                     auto actionAndArgs = winrt::make_self<winrt::TerminalApp::implementation::ActionAndArgs>();
                     actionAndArgs->Action(ShortcutAction::NewTab);
                     auto newTabArgs = winrt::make_self<winrt::TerminalApp::implementation::NewTabArgs>();
-                    newTabArgs->ProfileIndex(profileIndex);
+                    auto newTerminalArgs = winrt::make_self<winrt::TerminalApp::implementation::NewTerminalArgs>();
+                    newTerminalArgs->ProfileIndex(profileIndex);
+                    newTabArgs->TerminalArgs(*newTerminalArgs);
                     actionAndArgs->Args(*newTabArgs);
                     profileKeyChord = keyBindings.GetKeyBindingForActionWithArgs(*actionAndArgs);
```

Comment:
This is super expensive, we'll want to maybe not have it be so expensive later.

Reply:
(One can file a task for this)

Reply:
I thought `make_self` was the non-expensive version that just makes a `com_ptr` for the class, but didn't do the binary boundary hopping

Reply:
no no it is, but it's still a heap alloc of like four objects for what amounts to a hash table lookup :smile: